src/index.js

```diff
@@ -196,11 +198,36 @@ function sanitizeHtml(html, options, _recursing) {
             return;
           }
           var parsed;
+          // check allowedAttributesMap for the element and attribute and modify the value
+          // as necessary if there are specific values defined.
+          var passedAllowedAttributesMapCheck = false;
           if (!allowedAttributesMap ||
-              (has(allowedAttributesMap, name) && allowedAttributesMap[name].indexOf(a) !== -1 ) ||
-              (allowedAttributesMap['*'] && allowedAttributesMap['*'].indexOf(a) !== -1 ) ||
-              (has(allowedAttributesGlobMap, name) && allowedAttributesGlobMap[name].test(a)) ||
-              (allowedAttributesGlobMap['*'] && allowedAttributesGlobMap['*'].test(a))) {
+            (has(allowedAttributesMap, name) && allowedAttributesMap[name].indexOf(a) !== -1 ) ||
+            (allowedAttributesMap['*'] && allowedAttributesMap['*'].indexOf(a) !== -1 ) ||
+            (has(allowedAttributesGlobMap, name) && allowedAttributesGlobMap[name].test(a)) ||
+            (allowedAttributesGlobMap['*'] && allowedAttributesGlobMap['*'].test(a))) {    
+              passedAllowedAttributesMapCheck = true;
+          } else if (allowedAttributesMap && allowedAttributesMap[name]) {
+            for (const o of allowedAttributesMap[name]) {
+              if (isPlainObject(o) && o.name && (o.name === a)) {
+                passedAllowedAttributesMapCheck = true;
+                // now need to verify the value is allowed
+                const splitStrArray = value.split(' ');
```

Comment:
Hmmm. I see that this "we are actually allowing any combination of the allowed values separated by spaces" feature is useful for more attributes than just "class", but it is not documented and it could be a bad thing with attributes where it does not make sense. We need an additional property in the object to indicate this is what we want. `multiple: true` sounds good to me. When it is not present the entire value must match.

Reply:
Added an example and included a note to say that to match an entire value simply have one string in the array:

### "What if I want to allow only specific values on some attributes?"

When configuring the attribute in `allowedAttributes` simply use an object with attribute `name` and an allowed `values` array, this will allow any combination of the `values` in the array where the `values` are separated by a space. In the following example `sandbox="allow-forms allow-modals allow-orientation-lock allow-pointer-lock allow-popups allow-popups-to-escape-sandbox allow-scripts"` would become `sandbox="allow-popups allow-scripts"`:

```js
        allowedAttributes: {
          iframe: [ 
            {
              name: 'sandbox',
              values: ['allow-popups', 'allow-same-origin', 'allow-scripts']
            }
          ]
```
Note: If you only want to allow a single specific attribute value, then `values` should only have a single string in the array.